### PR TITLE
`General`: Let exercise and lecture list rows behave as links

### LINKED ISSUE #2820: Make Courses / Exercises work like links
### Is your feature request related to a problem?
Artemis 4.9.4 now thankfully has buttons which work like links (#611, #2669). But this functionality is missing for courses in the Course Overview and for exercises of a course. Opening multiple exercises in multiple browser tabs still requires to manually navigate through Artemis over and over.

### Describe the solution you'd like
Courses and Exercises also have link-like features (open in new Tab, middle mouse button).

### Describe alternatives you've considered
Leave everything as it is ¯\_(ツ)_/¯

### PR DESCRIPTION
**Pull request by coaxial1**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/).
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.

### Motivation and Context

Exercise and lecture list rows (student perspective) do not behave as link, but using JS navigate calls instead.
This PR let's them behave as links so you can open them in new tabs with a middle mouse click, for example.
Closes #2820.

### Description

Utilizing Boostraps stretched link class to overlay the existing rows with a real link tag.

### Steps for Testing

Prerequisites:
- 1 Student
- A course with exercises and lectures

1. Log in to Artemis
2. Navigate to Exercises
3. Verify that exercise row do have a real link on them (e.g. open in new tab to verify, use middle mouse click)
4. Verify that buttons on these rows still work as before and are *not* overlayed by the stretched link
5. Do the same in the lecture list

#### Code Review
- [ ] Review 1
- [ ] Review 2
#### Manual Tests
- [ ] Test 1
- [ ] Test 2

### Screenshots

https://user-images.githubusercontent.com/23171488/144301127-e66ed740-2359-45a8-acb5-de8ff2a0d30c.mov

### CODE REVIEW

**Review by ole-ve — APPROVED** [review]
Tested on TS1, works as described.
Code changes look also good.
Nice work, really appreciate the feature 👍

**Review by DanielKainz — APPROVED** [review]
I tested it on TS1. Everything works as expected. Great work  :-)

File: src/test/javascript/spec/component/lecture-unit/exercise-unit/exercise-unit.component.spec.ts [hunk 1]
```diff
@@ -67,7 +58,7 @@ describe('ExerciseUnitComponent', () => {
             By.directive(CourseExerciseRowStubComponent),
         ).componentInstance;
 
-        expect(courseExerciseRowStubComponent.course).to.equal(exerciseUnitComponent.course);
-        expect(courseExerciseRowStubComponent.exercise).to.equal(exerciseUnitComponent.exerciseUnit.exercise);
+        expect(courseExerciseRowStubComponent.course).toEqual(exerciseUnitComponent.course);
```
Comment (Malyuk-A) [inline src/test/javascript/spec/component/lecture-unit/exercise-unit/exercise-unit.component.spec.ts:61]: You replaced to.equal with to.Be in the other file, and due to our documentation I think it wouldn't be bad to stick to one, i.e. to.Be. What do you think?

**Review by Malyuk-A — APPROVED** [review]
Tested on TS1. Works great. (I am happy that this feature/issue is finally resolved 👍)
I have one minor nitpick with the tests. Otherwise, the code looks good to me.

**Review by baranimo7 — APPROVED** [review]
Tested on ts1, everything works as described.

**Review by MertAksehirlioglu — APPROVED** [review]
Tested it on TS1, works as expected.

File: src/test/javascript/spec/component/lecture-unit/exercise-unit/exercise-unit.component.spec.ts (hunk 1 above)
Comment (coaxial1) [reply src/test/javascript/spec/component/lecture-unit/exercise-unit/exercise-unit.component.spec.ts:61]: ``toBe`` and ``toEqual`` aren't the same thing; ``toBe`` works good for enum values, primitives and strings etc. as it is using ``Object.is(a, b)`` under the hood, but ``toEqual`` performs a deep quality check on objects, which is why it's used here.

**Pull request merged**